java/refactoring.java/src/org/netbeans/modules/refactoring/java/plugins/RenameRefactoringPlugin.java

```diff
@@ -433,6 +438,27 @@ public void run(CompilationController info) throws Exception {
                                     //add all references of overriding methods
                                     allMethods.add(ElementHandle.create((ExecutableElement)el));
                                 }
+                                if (kind == ElementKind.FIELD && el.getEnclosingElement().getKind() == ElementKind.RECORD) {
+                                    for (ExecutableElement m : ElementFilter.methodsIn(el.getEnclosingElement().getEnclosedElements())) {
+                                        if (m.getSimpleName().equals(el.getSimpleName()) && m.getParameters().isEmpty()) {
+                                            //accessor:
+//                                            addMethods(m, set, info, idx);
+                                            recordLinkedDeclarations.add(TreePathHandle.create(m, info));
+                                            addUsesOfMethod(m, set, info, idx);
+                                        }
+                                    }
+                                    for (ExecutableElement c : ElementFilter.constructorsIn(el.getEnclosingElement().getEnclosedElements())) {
+                                        if (!info.getElements().isCanonicalConstructor(c)) {
+                                            continue;
+                                        }
+                                        for (VariableElement param : c.getParameters()) {
+                                            if (param.getSimpleName().equals(el.getSimpleName())) {
+                                                //the parameter for the canonical constructor:
+                                                recordLinkedDeclarations.add(TreePathHandle.create(param, info));
+                                            }
+                                        }
+                                    }
+                                }
```

Comment:
L467 below would need the reverse case for renaming a record accessor which would have to be linked to the comonents i believe.